pkg/adapterManager/manager.go

```diff
@@ -390,13 +390,15 @@ func newCacheKey(kind config.Kind, cfg *cpb.Combined) (*cacheKey, error) {
 
 	if cfg.Builder.Params != nil {
 		if err := enc.Encode(cfg.Builder.Params); err != nil {
+			pool.PutBuffer(b)
 			return nil, err
 		}
 		ret.builderParamsSHA = sha1.Sum(b.Bytes())
 	}
 	b.Reset()
```

Comment:
I think the reset is needed because the bytes.Buffer is used by the encoder, which is still used below this point.